src/compiler/checker.ts

```diff
@@ -43324,7 +43324,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         getNodeLinks(node).flags |= NodeCheckFlags.LexicalModuleMergesWithClass;
                     }
                 }
-                if (compilerOptions.verbatimModuleSyntax && node.parent.kind === SyntaxKind.SourceFile) {
+                if (compilerOptions.verbatimModuleSyntax &&
+                    node.parent.kind === SyntaxKind.SourceFile &&
+                    (moduleKind === ModuleKind.CommonJS || getSourceFileOfNode(node).impliedNodeFormat === ModuleKind.CommonJS)

```

Comment:
Ignore this if you want (I don't think it matters), but having just checked `node.parent.kind` you could probably just do `node.parent.impliedNodeFormat`.

Reply:
Good call, I just copied and pasted the condition from the other place this error is issued.